openshift/caddy-reverse-proxy.dc.yaml

```diff
@@ -0,0 +1,119 @@
+---
+apiVersion: v1
+kind: Template
+labels:
+  template: "${APP_NAME}-reverse-proxy-template"
+  app: "${APP_NAME}-reverse-proxy"
+metadata:
+  name: "${APP_NAME}-reverse-proxy-dc"
+objects:
+  # - kind: NetworkSecurityPolicy
+  #   apiVersion: secops.pathfinder.gov.bc.ca/v1alpha1
+  #   metadata:
+  #     name: "${APP_NAME}-reverse-proxy-pods-to-egress-${NAMESPACE}"
+  #   spec:
+  #     description: |
+  #       Allow app pods to open connections to the internet
+  #     source:
+  #       - - "$namespace=${NAMESPACE}"
+  #         - "app=${APP_NAME}-reverse-proxy"
+  #         - "deploymentconfig=${APP_NAME}-reverse-proxy"
+  #         - role=reverse-proxy
+  #     destination:
+  #       - - ext:network=any
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: "${APP_NAME}-caddyfile"
+    data:
+      Caddyfile: |
+        *:2015 {
+            redir 302 {
+                https://comfort.pathfinder.gov.bc.ca/app/${APP_NAME}/
+            }
+
+            log / stdout "{combined}"
+
+            errors stdout
+        }
+  - apiVersion: v1
+    kind: DeploymentConfig
+    metadata:
+      name: "${APP_NAME}-reverse-proxy"
+    spec:
+      replicas: 2
+      selector:
+        app: "${APP_NAME}-reverse-proxy"
+        deploymentconfig: "${APP_NAME}-reverse-proxy"
+        role: reverse-proxy
+      strategy:
+        resources: {}
+        type: Rolling
+      template:
+        metadata:
+          labels:
+            app: "${APP_NAME}-reverse-proxy"
+            deploymentconfig: "${APP_NAME}-reverse-proxy"
+            role: reverse-proxy
+        spec:
+          containers:
+            - command:
+                - sh
+                - -c
+                - exec caddy -quic -conf /mnt/Caddyfile
```

Comment:
i think you want to explain this part in the readme, why you need to mount etc.

Reply:
Ok I'll add another section talking about that. The nutshell version is, the image we're using has a pre-baked Caddyfile already but doesn't serve our needs at all. As we only want to run Caddy with a custom caddyfile, I went the route of inserting the Caddyfile into the image via file volume mount. This allows us to avoid having to build an image for this, and instead be able to just run the base caddy image directly.

The custom command override is needed because the executable needs to know which caddyfile to use as configuration - in our case, we mounted it at /mnt/Caddyfile.

Reply:
yeah, I know that, but someone else won't, and it's very useful information if someone was just looking over the readme.

Reply:
Background added.